src/TrinaryLogic.php

```diff
@@ -28,22 +28,23 @@ private function __construct(private int $value)
 
 	public static function createYes(): self
 	{
-		return self::create(self::YES);
+		return self::$registry[self::YES] ??= new self(self::YES);
 	}
 
 	public static function createNo(): self
 	{
-		return self::create(self::NO);
+		return self::$registry[self::NO] ??= new self(self::NO);
 	}
 
 	public static function createMaybe(): self
 	{
-		return self::create(self::MAYBE);
+		return self::$registry[self::MAYBE] ??= new self(self::MAYBE);
```

Comment:
just theory crafting here so feel free to tell me if i'm wrong

but if phpstan is caching and reusing the same instance for `yes` `no` and `maybe`, would it be worth it to somehow pre-create the 3 instances needed and then return those directly? 

this would avoid millions of null checks, of which there are 3 cases where the null check is actually relevant (the first one of each case)

Reply:
It's a question of when it should be pre-precreated. PHPStan internals are not only executed through `bin/phpstan`, there are also other usages, like executing test cases, and also usage of internals in 3rd party tools like Rector.

Maybe it could work if we initialize it as a side effect right before `class TrinaryLogic` declaration.